src/kubelet/metric/metric.go

```diff
@@ -33,14 +33,12 @@ func GetMetricsData(c client.HTTPClient) (v1.Summary, error) {
 		return v1.Summary{}, fmt.Errorf("error reading the response body of kubelet endpoint. Got error: %v", err.Error())
 	}
 
-	var summary = new(v1.Summary)
-	err = json.Unmarshal(body, summary)
-	if err != nil {
+	summary := &v1.Summary{}
+	if err := json.Unmarshal(body, summary); err != nil {
 		return v1.Summary{}, fmt.Errorf("error unmarshaling the response body. Got error: %v", err.Error())
 	}
 
 	return *summary, nil
```

Comment:
```suggestion
	summary := v1.Summary{}
	if err := json.Unmarshal(body, &summary); err != nil {
		return v1.Summary{}, fmt.Errorf("error unmarshaling response into v1.Summary: %w", err)
	}

	return summary, nil
```

Reply:
Not a huge fan of returning empty objects rather than nil pointers though.

Reply:
>Not a huge fan of returning empty objects rather than nil pointers though.

Agreed. I've made a bit more improvements along the way, please have a look.